### PR TITLE
GitHub Actions for Elixir project

### PR DESCRIPTION
**Pull request by georgiybykov**
There is an [issue](https://github.com/actions/cache/issues/698) with caching in GitHub Actions.

So, when it is fixed we should to follow the instructions, that I have left inside `.github/workflows/ci.yml`, to start caching dependencies.

**p.s.** You are able to look at [this PR](https://github.com/georgiybykov/elixir-basic-project-template/pull/1) opened in the fork of the repo, to make sure that CI in GHA is working.

### CODE REVIEW

File: .dialyzer-ignore.exs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# The list of the files to ignoge could be placed below.
+#   SEE: https://github.com/jeremyjh/dialyxir#elixir-term-format
+[
+  {"lib/kv_behaviour.ex", :unknown_type}
```
Comment (am-kantox) [inline .dialyzer-ignore.exs:4]: We should not add entries here, we are supposed to fix issues instead :)

File: lib/key_value_store.ex [hunk 2]
```diff
@@ -1,6 +1,6 @@
 defmodule Kv do
   @moduledoc """
-  Key-Value store naïve implementation
+  Key-Value store naive implementation
```
Comment (am-kantox) [inline lib/key_value_store.ex:3]: Excuse me? The correct spelling is “naïve.”

**Review by am-kantox — APPROVED** [review]
I am to merge this and then fix the issue reported by dialyzer in a separate PR.

File: .dialyzer-ignore.exs (hunk 1 above)
Comment (georgiybykov) [reply .dialyzer-ignore.exs:4]: I am totally agree, however I made the same empty file for `dialyzer ignore` last time and did not check if it worked. So,  I decided to do it now :D

**Pull request merged**